File: DAM 25-26/DesarolloDeInterfaces/Back/SwapShop/src/main/java/com/example/SwapShop/modelos/IntercambiosPrestamos.java
package com.example.SwapShop.modelos;

import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;
import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "intercambios")
public class IntercambiosPrestamos {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    // ---- RELACIONES ----

    @ManyToOne
    @JoinColumn(name = "id_prenda", nullable = false)
    private Prendas prenda;

    @ManyToOne
    @JoinColumn(name = "id_prenda2", nullable = false)
    private Prendas prenda2;

    @ManyToOne
    @JoinColumn(name = "id_solicitante", nullable = false)
    private Usuario solicitante;

    @ManyToOne
    @JoinColumn(name = "id_dueno", nullable = false)
    private Usuario dueno;

    // ---- ENUMS ----

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoIntercambio tipo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EstadoIntercambio estado = EstadoIntercambio.pendiente;

    // ---- FECHAS ----

    @Column(name = "fecha_inicio")
    private Date fechaInicio;

    @Column(name = "fecha_fin")
    private Date fechaFin;
}

File: DAM 25-26/DesarolloDeInterfaces/Front/SwapShop/src/app/modelos/IntercambiosPrestamos.ts
import { Prenda } from './Prendas';
import { Usuario } from './Usuario';
import { EstadoIntercambio } from './estado-intercambio';
import { TipoIntercambio } from './tipo-intercambio';

export interface IntercambioPrestamos {
  id?: number;

  // Relaciones (solo IDs al crear)
  id_prenda: number;
  id_prenda2: number;
  id_solicitante: number;
  id_dueno: number;

  // Datos poblados en respuestas GET
  prenda?: Prenda;
  prenda2?: Prenda;
  solicitante?: Usuario;
  dueno?: Usuario;

  // Enums
  tipo: TipoIntercambio;
  estado: EstadoIntercambio;

  // Fechas (usamos string ISO para evitar problemas con java.sql.Date)
  fechaInicio?: string;    // "2025-12-20"
  fechaFin?: string;       // "2025-12-27"
}
